### DISCUSSION TITLE
Prowler Container Image on Codebuild

### DISCUSSION THREAD

**Discussion by NMuee** [prowler-cloud/prowler#1976] [Q&A]
Hello Prowler Team, 

I want to try using prowler container image to run in my codebuild. 
I reference public ECR prowler image. 
However, codebuild fail to provision with "Internal Service Error: CodeBuild is experiencing issues"

The same codebuild managed to run other image job. 

Wondering if there is config that I set up incorrectly? Or is it even possible to run prowler image in codebuild? 
Envrionment: Linux
Compute: 7 GB memory, 4 vCPUs

Thank you

**Comment by toniblyx** [comment]
Hi @NMuee we need more details, like what image, what version, how is that being deployed, etc. Make our task of helping you easier please.
Here is a sample template that make Prowler 3 works with CodeBuild https://github.com/prowler-cloud/prowler/blob/master/contrib/codebuild/codebuild-prowlerv3-audit-account-cfn.yaml

**Reply by NMuee** [reply]
I currently referencing public ECR (public.ecr.aws/o4g1s5r6/prowler:stable)
I tried directing codebuild custom image to this and also tried pulling this image to my own ECR repo and use my codebuild to reference to it. 

However, when i run my build, it failed at the provisioning phase. 

Thank you for the sample template, would definitely like to explore using the image instead of doing prowler install in the codebuild. Thank you.

**Reply by toniblyx** [reply]
As far as I know there are multiple reasons for CodeBuild to fail https://docs.aws.amazon.com/codebuild/latest/userguide/troubleshooting.html. Being provisioning phase it could be related to permissions. We use that very same image from other accounts with no issues.

**Reply by NMuee** [reply]
Thank you very much, I will continue to investigate and troubleshoot on my side.

**Reply by NMuee** [reply]
Hello @toniblyx , 
Wondering if you have any simple set up guide for using the prowler image in public ECR as custom image in codebuild? 
I just reached out to the AWS Support team but they also encounter the same issue. 
Wondering if there is any config or setting that i might missed out?

**Reply by toniblyx** [reply]
Is AWS Support unable to let you know what is the actual error?  "Internal Service Error: CodeBuild is experiencing issues" doesn't seem to help much to know what's going on. Weird.

Isn't it about the container architecture or similar? Have you tried building your own with our Dockerfile and using your own ECR to test? if that works it is probably a matter of permissions in your side? but we are pretty confident that others are using that image since it has almost 1M pulls.


We don't have such a set up guide.
